nshutijonathan/Auto-Mart
# #166373040 Implemented week1 lfa’s feedback on UI  in  Auto-Mart project

#### What does this PR do?
Implemented UI feedbacks 
Have documented API endpoints with swagger 
#### Description of Task to be completed?

- changed get started button on UI
- Added seller functionalities into one platform with the buyer
- Added get specific car functionalities 
- Corrected spelling errors
- Documented the API endpoint with swagger 
- added token based authorization in endpoints 
Have The following endpoint working and documented 
* POST/localhost:3000/api/v2/auth/signup
* POST/localhost:3000/api/v2/auth/signin
* POST/localhost:3000/api/v2/auth/signup/admin
* GET/localhost:3000/api/v2/users
*   GET/localhost:3000/api/v2/users/<:id>
*   DELETE/localhost:3000/api/v2/users/<:id>
*   POST/localhost:3000/api/v2/car
*   GET/localhost:3000/api/v2/cars
#### How should this be manually tested?
* Download or clone the tutorial project code from [this repo](https://github.com/nshutijonathan/Auto-Mart3), Install all required NPM  packages by running NPM install from the command line in the project root folder (where the package.json is located).
* After cloning the repo, cd into it then  open index.html in the browser or visit [Auto-Mart](https://nshutijonathan.github.io/Auto-Mart/html/) to see all the specified pages and implemented feedbacks on UI folder
* Start the API by running NPM  start from the command line in the project root folder, you should see the message Server listening on port 3000. You can test the API directly using an application such as Postman or you can test it with one of the single pages in the browser.
* You must have node environment configured  and .env file in the root folder
* to test authentication and authorization, add this to the header 
Key: :x-auth-token,value:JWT TOKEN
#### Any background context you want to provide?
* There is Implementation of Javascript(ES6) logic with s data structures  to persist some data
*Swagger to document API endpoints 
* Implemented CSS to implement feedback 
#### What are the relevant pivotal tracker stories?
[#166373040](https://www.pivotaltracker.com/n/projects/2346053/stories/166373040)

#### Screenshots
####Questions
--
